### PR TITLE
Improving core allocation experiment readme

### PR DESCRIPTION
**Pull request by iwankgb**
Fixes core allocation experiment readme

Summary of changes:
- updated readme.

Testing done:
- none.

### CODE REVIEW

File: experiments/optimal-core-allocation/README.md [hunk 1]
```diff
@@ -2,32 +2,32 @@
 
 ## Rationale
 
-This experiment allows you to find the optimal core and thread workers configuration that delivers the best performance 
-in terms of throughput and guaranteeing latency for sensitive workloads.
+This experiment allows you to find the optimal core and worker threads configuration that delivers the best performance 
+in terms of throughput and guaranteeing response time for latency sensitive workloads.
 
-This example experiment uses memcached as the latency sensitive workload and tries to find optimal number of memcached worker threads (typically set with  the `-t` flag for `memcached`).
+The experiment uses memcached as the latency sensitive workload and tries to find optimal number of memcached worker threads (typically set with  the `-t` flag for `memcached`).
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: Does experiment have any intelligence so that it can find anything? Maybe we should rephrase it to "helps user to find optimal (...)"

File: experiments/optimal-core-allocation/README.md [hunk 2]
```diff
@@ -2,32 +2,32 @@
...
 
-#### Environment prerequisites
+### Environment prerequisites
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: Should we write that all those things are available in "prepare_experiment_environment.sh"?

File: experiments/optimal-core-allocation/README.md [hunk 3]
```diff
@@ -37,15 +37,14 @@ where `number of maximum threads dedicated to memcached` is by default set to nu
 sudo optimal-core-allocation -experiment_peak_load=800000
 ```
 
-2. To print results from the experiment, you need to copy `experiment id` in form of a UUID e.g. `ca24aa5d-4a88-7258-6f00-5f651b0d6515`.
-The ID is printed to the standard output and should be used in python functions to generate result tables in Jupyter, like this:
+2. To display results of the experiment, you need to copy `experiment id` (eg `ca24aa5d-4a88-7258-6f00-5f651b0d6515`) and pass it as a parameter to python functions on Jupyter notebook:
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: There is no way told how to get Experiment_ID. 
For example, in main readme I have written `sudo memcached-sensitivity-profile (...) > uuid.txt` so that user can get uuid without following logs.

File: experiments/optimal-core-allocation/README.md [hunk 4]
```diff
@@ -60,28 +59,28 @@ core.cpu() # if run with USE snap collector
...
 
-This example shows how to run the experiment using configuration file.
-All experiment flags can be provided using configuration file. Command line flag `-foo_bar` is equal to `FOO_BAR` option in configuration file.
+This example shows how to run the experiment using a configuration file.
+All experiment flags can be provided using a configuration file. Command line flag `-foo_bar` is equal to `FOO_BAR` option in configuration file.
 
 1. Generate default configuration: 
 
 ```bash
 sudo optimal-core-allocation -config-dump >example-configuration.txt
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: Our conf file acutally follows `ini` format, so I have started to use it in Sensitivity Profile Documentation (e.g. `config.ini`). This way I got syntax coloring :) What do you think about that?

File: experiments/optimal-core-allocation/README.md [hunk 5]
```diff
@@ -60,28 +59,28 @@ core.cpu() # if run with USE snap collector
...
-3. Run experiment with provided configuration:
+3. Run experiment with your configuration:
 
 ```bash
-sudo optimal-core-allocation -config example-configuration.txt -experiment_peak_load=800000
+sudo optimal-core-allocation -config example-configuration.txt
 ```
 
-4. (Optionally) Run experiment with override configuration flags.
+4. Run the experiment overriding configuration file values using flags:
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: This could be marked as (optional) as an example that you actually *can* override things. What would you say?

File: experiments/optimal-core-allocation/README.md [hunk 6]
```diff
@@ -98,96 +97,88 @@ where:
 
 #### Other experiments variants
 
-1. You can run this experiment with memcached processes pinned to specified number of hardware threads (`-use-core-pinning` flag).
-1. You can run memcached with a patch responsible for pinning worker threads to specified processors (flag: `-memcached_threads_affinity` flag).
-1. You can run memcached deployed by Kubernetes (`-kubertenes` flag). 
+1. You can run the experiment with memcached threads pinned to specified number of hardware threads (`-use-core-pinning` flag).
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: Later we should elaborate how this differs from memcached affinity :)

File: experiments/optimal-core-allocation/README.md [hunk 7]
```diff
@@ -98,96 +97,88 @@ where:
...
+1. You can run the experiment with memcached patch that allows to pin worker threads to single CPU (flag: `-memcached_threads_affinity` flag).
+1. You can run the experiment on Kubernetes cluster (`-kubertenes*` flags).
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: We can add "Swan will take care of setting up the cluster".

File: experiments/optimal-core-allocation/README.md [hunk 8]
```diff
@@ -98,96 +97,88 @@ where:
...
 - 9 nodes for load generator cluster (1 master node and 8 agents),
 - Linux distribution: CentOS 7 with 4.10 Linux kernel,
 
 ### How to read results
 
-The tables below show how number of worker threads are able to handle provided load.
+The tables below show how memcached capacity changes when amount of worker threads grows.
 
-Each cell shows 99% percentile latency, i.e. tail latency, with colors indicating violation of stated SLO.
+Each cell displays 99% percentile latency, i.e. tail latency. Colors indicates violation (or lack of thereof) of SLO:
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: /s/Each cell displays 99% percentile latency, i.e. tail latency/Each cell displays Memacahed tail latency (99 percentile).

File: experiments/optimal-core-allocation/README.md [hunk 9]
```diff
@@ -98,96 +97,88 @@ where:
...
+- yellow - tail latency between 101% and 150% of SLO, 
+- red - tail latency above 150% of SLO, 
+- gray - memcached was incapable of handling requested amount of QPS.
 
 There are two dimensions:
 - load (x axis) - fraction of peak load expressed in QPS (Queries Per Second)
 - worker threads (y axis) - number of memcached worker threads (check memcached -t option for details).
 
-Results below were run with 10 load points with peak load set to 1.5 million of QPS with SLO set to 500 us.
+The experiment has been run with 10 load points and peak load set to 1.5 million QPS, and SLO set to 500 us.
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: Should we swap first "and" with comma?

File: experiments/optimal-core-allocation/README.md [hunk 10]
```diff
@@ -98,96 +97,88 @@ where:
...
-After increasing the accepted tail latency, i.e. SLO, to 3ms there are additional acceptable configurations we can choose from.
-We can assign less threads which are capable to handle even more load, while still guaranteeing that our new relaxed SLO is met.
+After increasing the accepted tail latency to 3ms there we can choose from broader range of configurations.
+Less threads are necessary to handle requests, while relaxed target SLO is met.
 
-In this case optimal configuration can use **6, 7 or 8 threads**, depending of desired machine utilization, which should be capable of handling more than 1m QPS and with response below 3ms.
+In this case optimal configuration will utilize **7 threads** and will be capable of handling more than 1.000.000 QPS with tail latency below 3ms.
 
-##### Discussion
+#### Discussion
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: @ppalucki it would be awsome if we could draw a graph comparing two experiments here :)

**Comment by ppalucki** [comment]
run integration tests

File: experiments/optimal-core-allocation/README.md (hunk 1 above)
Comment (iwankgb) [inline experiments/optimal-core-allocation/README.md]: @skonefal - I don't think so. It would have to be copied to all the READMEs.

File: experiments/optimal-core-allocation/README.md (hunk 2 above)
Comment (iwankgb) [inline experiments/optimal-core-allocation/README.md]: @skonefal - I don't think so. It would have to be copied to all the READMEs.

File: experiments/optimal-core-allocation/README.md [hunk 11]
```diff
@@ -2,50 +2,49 @@
...
+The experiment helps user to find optimal number of memcached worker threads (typically set with  the `-t` flag for `memcached`) by running memcached as the latency sensitive workload with various worker threads configuration.
 
-#### Environment prerequisites
+### Environment prerequisites
 
-1. Download and install in $PATH `optimal-core-allocation` experiment binary from [releases page](https://github.com/intelsdi-x/swan/releases) (available inside `swan.tar.gz`) along with all required [snap](https://github.com/intelsdi-x/snap) plugins.
-1. [mutilate](https://github.com/leverich/mutilate) load generator preinstalled on load generator nodes (flags: `-experiment_mutilate_master_address`, `-experiment_mutilate_agent_addresses`).
+1. `optimal-core-allocation` experiment binary available in `$PATH`. The binary can be downloaded from [releases page](https://github.com/intelsdi-x/swan/releases) (available inside `swan.tar.gz`) along with all required [Snap](https://github.com/intelsdi-x/snap) plugins.
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: Not all snap plugins are in package. This experiment can use USE collector that is not available in this package.
Could you please point to `https://github.com/intelsdi-x/snap-plugin-collector-use/releases/download/1/snap-plugin-collector-use_linux_x86_64` to download it?

File: experiments/optimal-core-allocation/README.md [hunk 12]
```diff
@@ -2,50 +2,49 @@
...
+1. [mutilate](https://github.com/leverich/mutilate) load generator installed on load generator nodes (flags: `-experiment_mutilate_master_address`, `-experiment_mutilate_agent_addresses`).
 1. Running Cassandra database with a keyspace dedicated to swan to store the experiment results (flag: `-cassandra_address`).
-1. Jupyter notebook with `jupyter/generic.py` module installed (with dependencies installed from `jupyter/requirements.txt`).
-1. (optional for distributed mode) ssh password-less access to load generator nodes (flags: `-remote_ssh_key_path`, `-remote_ssh_login`).
+1. Runnin Jupyter notebook server (use [`intelsdi/swan-jupyter`](https://hub.docker.com/r/intelsdi/swan-jupyter/) Docker image.
```
Comment (skonefal) [inline experiments/optimal-core-allocation/README.md]: Drop runnin or add 'g' :)
Add ")" before Docker.
